src/video-recorder/recorder.js

```diff
@@ -132,13 +132,18 @@ export default class VideoRecorder extends EventEmitter {
         const testRunVideoRecorder = this._createTestRunVideoRecorder(testRunInfo, recordingOptions);
         const isVideoSupported     = await testRunVideoRecorder.isVideoSupported();
 
-        if (isVideoSupported) {
-            await testRunVideoRecorder.init();
-
-            this.testRunVideoRecorders[testRunVideoRecorder.index] = testRunVideoRecorder;
-        }
-        else
+        if (!isVideoSupported) {
             this.warningLog.addWarning(WARNING_MESSAGES.videoNotSupportedByBrowser, testRunVideoRecorder.testRunInfo.alias);
+            return;
+        }
+
+        const isVideoEnabled = await testRunVideoRecorder.isVideoEnabled();
+
+        if (!isVideoEnabled)
+            return;
+
+        await testRunVideoRecorder.init();
```

Comment:
It's better to separate `await` statements from synchronous code with empty lines